inst/tests/tests.Rraw

```diff
@@ -14977,6 +14977,67 @@ DT = data.table(
 test(2054, DT[order(C)[1:5], B, verbose=TRUE], c('b', 'b', 'c', 'c', 'a'),
            output = 'order optimisation is on')
 
+# #3424 -- coalesce function
+bool = c(TRUE, NA, FALSE)
+bool_val = c(TRUE, TRUE, FALSE)
+int = c(1L, 2L, NA_integer_, 4L)
+int_val = 1:4
+num = c(1, 2, NA_real_, 4)
+num_val = c(1, 2, 3, 4)
+str = c('a', NA_character_, 'b', NA_character_)
+str_val = c('a', 'b', 'b', 'b')
+fkt = factor(str)
+fkt_val = factor(str_val)
+
+## singleton replacements
+test(2055.01, coalesce(bool, TRUE), bool_val)
+test(2055.02, coalesce(bool, NA, TRUE), bool_val)
+test(2055.03, coalesce(int, 3L), int_val)
+test(2055.04, coalesce(int, NA_integer_, 3L), int_val)
+test(2055.05, coalesce(num, 3), num_val)
+test(2055.06, coalesce(num, NA_real_, 3), num_val)
+test(2055.07, coalesce(str, 'b'), str_val)
+test(2055.08, coalesce(str, NA_character_, 'b'), str_val)
+test(2055.09, coalesce(fkt, factor('b', levels = c('a', 'b'))), fkt_val)
+test(2055.10, coalesce(fkt, NA_integer_, factor('b', levels = c('a', 'b'))), fkt_val)
+
+## vector replacements
+test(2055.11, coalesce(bool, rep(TRUE, 3L)), bool_val)
+test(2055.12, coalesce(bool, rep(NA, 3L), rep(TRUE, 3L)), bool_val)
+test(2055.13, coalesce(int, rep(3L, 4L)), int_val)
+test(2055.14, coalesce(int, rep(NA_integer_, 4L), rep(3L, 4L)), int_val)
+test(2055.15, coalesce(num, rep(3, 4L)), num_val)
+test(2055.16, coalesce(num, rep(NA_real_, 4L), rep(3, 4L)), num_val)
+test(2055.17, coalesce(str, rep('b', 4L)), str_val)
+test(2055.18, coalesce(str, rep(NA_character_, 4L), rep('b', 4L)), str_val)
+test(2055.19, coalesce(fkt, factor(rep('b', 4L), levels = c('a', 'b'))), fkt_val)
+test(2055.20, coalesce(fkt, rep(NA_integer_, 4L), factor(rep('b', 4L), levels = c('a', 'b'))), fkt_val)
+
+## .dots argument
+test(2055.21, coalesce(bool, .dots = list(NA, TRUE)), bool_val)
+
+## edge cases/checks
+test(2055.22, coalesce(bool), bool)
+test(2055.23, coalesce(fkt), fkt)
+test(2055.24, coalesce(bool, 1L), error = 'internal type integer')
+#test(2055.25, coalesce(bool, NA_integer_), error = 'internal type integer')
+#test(2055.26, coalesce(fkt, 1L), error = '')
+#test(2055.27, coalesce(num, 3L), error = 'internal type integer')
+#test(2055.28, coalesce(int, 3), error = 'internal type double')
```

Comment:
two more placeholder tests -- do we really want to error here? or worth to add `isReallyReal`-type tests?

`numeric`->`integer` should always be doable right?

Reply:
I lean towards error, requesting user to provide proper type.
For the sake of keeping code less complex, and preaching good practices